tests/test_types.py

```diff
@@ -83,3 +83,31 @@ def test_mutating_should_not_optimize(mocked_optimizer):
     assert result
     assert result.pk == 7
     assert mocked_optimizer.call_count == 0
+
+
+@pytest.mark.django_db
+@patch('graphene_django_optimizer.types.query',
+       return_value=SomeOtherItem.objects)
+def test_should_optimize_the_queryset(mocked_optimizer):
+    SomeOtherItem.objects.create(pk=7, name='Hello')
```

Comment:
I think it's not a good idea to fix a PK here, I would prefer:
```py
some_other_item = SomeOtherItem.objects.create(name='Hello')

# ...

qs = SomeOtherItem.objects.filter(pk=some_other_item.pk)

# ...

assert result.pk == some_other_item.pk
```

On the other hand, you went right into the point on this test and that's fine because that's the idea when testing. But I think you also should cover the `get_node` method because that's why @tfoxy wrote the code in the first place and why I had the issue. So, I think is better to also add a test for the use case.

Reply:
The other tests have a fixed PK or id, so I believe that's why this was also done here. Not sure if there's a reason to have that as a fixed value (I didn't write this code). Maybe there's something with auto-incrementing ids that can give false positives.

About `get_node`, the other tests in this file are asserting results by calling that directly. Are those enough or do you believe there are some cases that are not covered here?

I don't really have a problem merging this unless you see there's something that can potentially break someones code. The only thing that I see that it can break is people not expecting `get_queryset` to optimize, but I believe that's not an issue.

Reply:
Hi @tfoxy!

> The other tests have a fixed PK or id, so I believe that's why this was also done here. Not sure if there's a reason to have that as a fixed value (I didn't write this code). Maybe there's something with auto-incrementing ids that can give false positives.

The issue with fixed IDs is a possible collision when adding new tests/fixtures or running them with `--reuse-db` flag. On the other hand, it's not a big deal but I prefer to not fix them. I understand other tests also do that but I think is better to start not doing it anymore. Anyway, it's not a blocker and if an issue appears in the future it can be fixed when it happens.

> About `get_node`, the other tests in this file are asserting results by calling that directly. Are those enough or do you believe there are some cases that are not covered here?

It's hard for me to think about fixing an issue without adding a test to make it fails in the first place. On the other hand, the code is covered by other tests, I agree with that. Anyway, I'd like to add at least one more test to the PR if @maxpeterson is ok with that.

> I don't really have a problem merging this unless you see there's something that can potentially break someones code. The only thing that I see that it can break is people not expecting get_queryset to optimize, but I believe that's not an issue.

I think it would be very difficult to break someone's code because Django already takes care of that. If you use `.only()` in a query set and then try to get a property not included in the call, Django will perform another query to get the value, also in the templates.

Reply:
Yes, now you make me remember that I had problems with fixed ids in tests in the past. I agree with that.

About the tests, yes, I wouldn't complain about having more tests. But still, @maxpeterson if you don't have enough time or not sure what to do, feel free to say so and I will merge it as it is. Minimum change for this PR would be the pk change as @iorlandini posted in the first comment.

Reply:
Sorry for the slow response. I am also not keen on hard coded ids - I normally use https://factoryboy.readthedocs.io but in this case I kept the tests as close to the others.

@iorlandini I think I have missed something. What additional test do you want / want to cover. I didn't add the tests for the case in #57 as I thought it was probably best handled in a separate PR.

I don't have a lot of spare time, but happy to make changes as needed when I can fit them in.